### ISSUE TITLE
Volumes created with 1.3.1 crash the Finder

### ISSUE THREAD

**Issue by mbarringer** [al45tair/dmgbuild#7]
I upgraded our build server from 10.11 to 10.12 and ran into a problem with dmgbuild 1.2.1 that we were using, so I upgraded it to 1.3.1 and python 3. 

The DMGs are created correctly, and the volumes _mount_ correctly, but the volumes that it produces causes Finder to crash when I try to open them (tested on two 10.12.6 machines):

> *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFData getCString:maxLength:encoding:]: unrecognized selector sent to instance 0x610000249db0'
> terminating with uncaught exception of type NSException
> abort() called
>
> Application Specific Backtrace 1:
> 0   CoreFoundation                      0x00007fffd13422cb __exceptionPreprocess + 171
> 1   libobjc.A.dylib                     0x00007fffe615a48d objc_exception_throw + 48
> 2   CoreFoundation                      0x00007fffd13c3f04 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
> 3   CoreFoundation                      0x00007fffd12b4755 ___forwarding___ + 1061
> 4   CoreFoundation                      0x00007fffd12b42a8 _CF_forwarding_prep_0 + 120
> 5   Foundation                          0x00007fffd2cfb9b8 parseFloats + 89
> 6   Foundation                          0x00007fffd2cfb939 NSRectFromString + 59
> 7   Finder                              0x000000010589e57f Finder + 2745727
> 8   Finder                              0x000000010566150c Finder + 398604
> ...

 
This occurs both with my previously working setup, as well as the volumes that 'python3 setup.py test' produces, in both the released 1.3.1 tarball as well as master. How can I start tracking this problem down?

**Comment by al45tair** [comment]
On 22 Sep 2017, at 13:22, Matt Barringer <notifications@github.com> wrote:
> 
> *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFData getCString:maxLength:encoding:]: unrecognized selector sent to instance 0x610000249db0'
> terminating with uncaught exception of type NSException
> abort() called

If I had to guess, from the backtrace, this is a Python 3 problem; quite likely we’re sending a “bytes” object rather than an “str” object to the biplist code, which is causing it to generate an NSData instead of an NSString in the binary plist.

Have you tried using Python 2 instead?

Kind regards,

Alastair.

--
http://alastairs-place.net

**Comment by mbarringer** [comment]
With python 2.7.13 installed globally via Homebrew, I have a different error, which was why I switched to python 3:

```
E   internal_error: PyObjCRT_SizeOfType: Unhandled type '0x75', u'f', u'd']
```
That is despite uninstalling 2.7.13 after the OS upgrade and then reinstalling.

But it turns out that running dmgbuild within a Python 2.7.13 **_virtualenv_** works perfectly, so I'll close this. Sorry for the trouble!

**Comment by mikedarcy** [comment]
I am encountering this issue as well on Python3, and using Python 2.7.13 is not really a practical option for me at the moment, since the invocation of dmgbuild is part of a larger build process that goes on in a Buildbot virtualenv for a Python3 app.

If dmgbuild is going to support Python3 compatibility then this issue should probably be re-opened since it seems like a legitimate bug.

**Comment by al45tair** [comment]
Agreed, this shouldn't have been closed. I already have what I hope is a fix, which I've just checked in; I'll push out a release if it works (I'm just fixing the CI set-up because pyenv's maintainer appears to have pulled it from PyPI…)

**Comment by al45tair** [comment]
OK. This is fixed in dmgbuild 1.3.2.
